src/lib/yang/alarms.lua

```diff
@@ -389,10 +402,17 @@ end
 -- flag.
 local function update_alarm (key, alarm, args)
    if needs_status_change(alarm, args) then
+      local alt_resource = args.alt_resource or alarm.alt_resource or {}
+      assert(type(alt_resource) == "table")
+
       local status = {
          time = assert(format_date_as_iso_8601()),
          perceived_severity = assert(args.perceived_severity or alarm.perceived_severity),
          alarm_text = assert(args.alarm_text or alarm.alarm_text),
+         resource      = key.resource,
+         alarm_type_id = key.alarm_type_id,
+         alarm_type_qualifier = key.alarm_type_qualifier,
+         alt_resource = alt_resource,
```

Comment:
Same as before, this change is not correct because doesn't respect the semantics of the YANG schema.